sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -4133,4 +4133,17 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       origin = inlineTable.origin
     )
   }
+
+  def ordinalOutOfBoundsError(
+      ordinal: Int,
+      attributes: Seq[Attribute]): Throwable = {
+    new AnalysisException(
+      errorClass = "COLUMN_ORDINAL_OUT_OF_BOUNDS",
+      messageParameters = Map(
+        "ordinal" -> ordinal.toString,
+        "attributesLength" -> attributes.length.toString,
+        "attributes" -> attributes.map(_.name).mkString(", ")
```

Comment:
since attribute names are ids, please, wrap them by `toSQLId()`

Reply:
Please check if it is ok now. Thanks